dim/src/scanners/base.rs

```diff
@@ -100,18 +102,23 @@ impl MetadataExtractor {
         };
 
         let target_file_clone = target_file.clone();
-        let res = {
+        let mf_id = {
             let mut tx = self
                 .conn
                 .read()
                 .begin()
                 .await
                 .map_err(|e| ScannerError::DatabaseError(format!("{:?}", e)))?;
 
-            MediaFile::get_by_file(&mut tx, &target_file_clone).await
+            MediaFile::get_by_file(&mut tx, &target_file_clone)
+                .await
+                .map(|mf| mf.id)
+                .ok()
```

Comment:
This might cause trouble. I think we will have to modify the MediaFile query to return `Result<Option<....>, Error>` or we could match here with `matches` to check if the error is because no rows were found, or the database has other issues.